core/domain/fs_domain.py

```diff
@@ -471,6 +550,64 @@ def isfile(self, filepath):
         except cloudstorage.NotFoundError:
             return False
 
+    def get_file_content(self, filepath):  # pylint: disable=unused-argument
+        """Gets the content of the file stored in GCS
+        Args:
+            filepath: str. The path to the relevant file within the exploration.
+
+        Returns:
+            Content. The content of the file.
+        """
+        bucket_name = app_identity_services.get_gcs_resource_bucket_name()
+        gcs_file_url = (
+            '/%s/%s/assets/%s' % (
+                bucket_name, self._exploration_id, filepath))
+        gcs_file = cloudstorage.open(gcs_file_url)
+        contents = gcs_file.read()
+        gcs_file.close()
+
+        return contents
+
+    def get_image_dimensions(self, filepath):
+        """Gets the dimensions of the image with the given filename
+
+        Args:
+            filepath: str. The path to the relevant file within the exploration.
+
+        Returns:
+            tuple. Returns height and width of the image.
+        """
+        content = self.get_file_content(filepath)
+        height, width = gae_image_services.get_image_dimensions(content)
+        return height, width
+
+    def create_compressed_versions_of_image(self, filepath):
```

Comment:
Same thing. Have an image_services file that proxies to platform/

Reply:
So by this you mean --- I shall create a core/domain/image_services that calls the gae_image_service in the platform and does the compression. So wherever I require the compression of images i will just call the `core/domain/image_service` that would use the `core/platform/image/gae_image_service` to compress the image content and `core/domain/fs_domain.py` for commiting the new compressed versions to the system.

Let me know If it's correct or not?